src/routes.c

```diff
@@ -437,6 +447,92 @@ static wolfsentry_errcode_t wolfsentry_route_init(
+    struct wolfsentry_route *new)
+{
+    if (route_exports->private_data_size > data_addr_offset)
+        WOLFSENTRY_ERROR_RETURN(INVALID_ARG);
+    if (data_addr_size < WOLFSENTRY_BITS_TO_BYTES(route_exports->remote.addr_len) + WOLFSENTRY_BITS_TO_BYTES(route_exports->local.addr_len))
+        WOLFSENTRY_ERROR_RETURN(BUFFER_TOO_SMALL);
+    if (data_addr_size > MAX_UINT_OF(uint16_t))
+        WOLFSENTRY_ERROR_RETURN(NUMERIC_ARG_TOO_BIG);
+    if ((unsigned)data_addr_offset > MAX_UINT_OF(uint16_t))
+        WOLFSENTRY_ERROR_RETURN(NUMERIC_ARG_TOO_BIG);
+    if (! (route_exports->flags & (WOLFSENTRY_ROUTE_FLAG_DIRECTION_IN | WOLFSENTRY_ROUTE_FLAG_DIRECTION_OUT)))
+        WOLFSENTRY_ERROR_RETURN(INVALID_ARG);
+
+    /* make sure wildcards are sensical. */
+    if (((route_exports->flags & WOLFSENTRY_ROUTE_FLAG_SA_FAMILY_WILDCARD) &&
+         ((! (route_exports->flags & WOLFSENTRY_ROUTE_FLAG_SA_REMOTE_ADDR_WILDCARD)) ||
+          (! (route_exports->flags & WOLFSENTRY_ROUTE_FLAG_SA_LOCAL_ADDR_WILDCARD)) ||
+          (! (route_exports->flags & WOLFSENTRY_ROUTE_FLAG_SA_PROTO_WILDCARD)))) ||
+        ((route_exports->flags & WOLFSENTRY_ROUTE_FLAG_SA_PROTO_WILDCARD) &&
+         ((! (route_exports->flags & WOLFSENTRY_ROUTE_FLAG_SA_REMOTE_PORT_WILDCARD)) ||
+          (! (route_exports->flags & WOLFSENTRY_ROUTE_FLAG_SA_LOCAL_PORT_WILDCARD)))))
+        WOLFSENTRY_ERROR_RETURN(INVALID_ARG);
+
+    memset(new,0,offsetof(struct wolfsentry_route, data));
+
+    new->parent_event = parent_event;
+    new->flags = route_exports->flags;
+    new->sa_family = route_exports->sa_family;
+    new->sa_proto = route_exports->sa_proto;
+    new->remote.sa_port = route_exports->remote.sa_port;
+    new->remote.addr_len = route_exports->remote.addr_len;
+    new->remote.interface = route_exports->remote.interface;
+    new->local.sa_port = route_exports->local.sa_port;
+    new->local.addr_len = route_exports->local.addr_len;
+    new->local.interface = route_exports->local.interface;
+    new->data_addr_offset = (uint16_t)data_addr_offset;
+    new->data_addr_size = (uint16_t)data_addr_size;
+
+    if (data_addr_offset > 0) {
+        if (route_exports->private_data != NULL) {
+            memcpy(new->data, route_exports->private_data, route_exports->private_data_size); /* copy private data. */
+            if ((size_t)data_addr_offset > route_exports->private_data_size)
+                memset(new->data + route_exports->private_data_size, 0, (size_t)data_addr_offset - route_exports->private_data_size); /* zero leftovers. */
+        } else
+            memset(new->data, 0, (size_t)data_addr_offset); /* zero private data. */
+    }
+
+    new->meta.purge_after = route_exports->meta.purge_after;
+    new->meta.connection_count = route_exports->meta.connection_count;
+    new->meta.derogatory_count = route_exports->meta.derogatory_count;
+    new->meta.commendable_count = route_exports->meta.commendable_count;
+
+    if (route_exports->remote.addr_len > 0)
```

Comment:
can you adjust this section for readability? Having {} around the if statements when so close to a statement {} on its own, along with limiting it to 80 columns

Reply:
good call on the curlies for clarity.  I fixed the line lengths too, but some day I'm going to have to do a full code traversal to fix the rest -- there's 2106 lines over 80 chars wide!